src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1620,21 +1620,57 @@ public static int getConcurrentReaders()
         return conf.concurrent_reads;
     }
 
+    public static void setConcurrentReaders(int concurrent_reads)
```

Comment:
What's the intended purpose of these methods?

Reply:
It was to keep the configuration up to date with the system. I don't think it's as important as it was but when discussing with @clohfink he thought it was still valuable to keep the config in sync. 

The getters for the writers like `getConcurrentWriters` are called from `PaxosState`/`ViewManager` to size their `LOCKS` however I thought that given the operational use case for this is reducing concurrency from the starting value the locks will have been initialized large enough.

Reply:
They look to be used in a fairly imprecise manner, so even if we increase the number of threads I don't expect it should massively worsen things.  Ass to whether or not we need to keep the config in sync - I guess there's a purity in that, but we could also rename our getters to indicate the value is one-use.  We certainly don't update the config in all places we modify behaviour that was seeded by a value from the config.  That's not to say we shouldn't start now, but it seems like a policy change worth raising up the flag pole.  I don't personally see _huge_ value in it, except perhaps when debugging a heap dump, but I don't usually go and look at the `Config` object then anyway.  That's all to say that it doesn't really matter too much to me either way, but I have a personal preference for simpler code where possible.

Reply:
Agreed it isn't uniformly updated and perfect, but keeping the Config updated with the current values may avoid confusion for debugging heap dumps as you mentioned. As you don't object, I'll keep the code as is.